### PR TITLE
SAK-46812 Assignments students can insert a timesheet without a comment

### PR DESCRIPTION
**Pull request by feralvarez95**
https://sakaiproject.atlassian.net/browse/SAK-46812

### CODE REVIEW

File: assignment/tool/src/webapp/js/studentViewSubmission.js [hunk 1]
```diff
@@ -107,45 +124,59 @@ ASN.switchTimesheetTab = function (source) {
...
     button.classList.remove("spinButton");
     button.disabled = false;
-    alert('Error: ' + xhr.status);
-    console.error("Ajax call error when add time sheet register.");
+    document.getElementById("alertTsheetAddRecord").style.display="block";
+    let messageArray = [];
+    for (const [index, key] of Object.entries(messages)) {
+        messageArray.push(window.i18nWlogTab[key]);
+    }
+    document.getElementById("alertTsheetAddRecord").innerHTML= messageArray.join('<br>');
```
Comment (adrianfish) [inline assignment/tool/src/webapp/js/studentViewSubmission.js:165]:
```suggestion
    document.getElementById("alertTsheetAddRecord").innerHTML
      = Object.entries(messages).reduce((acc, entry) => { acc.push(i18nWlogTab[entry[1]]); return acc; }, []).join("<br>");
```

File: assignment/tool/src/webapp/js/studentViewSubmission.js [hunk 2]
```diff
@@ -107,45 +124,59 @@ ASN.switchTimesheetTab = function (source) {
...
     button.classList.remove("spinButton");
     button.disabled = false;
-    alert('Error: ' + xhr.status);
-    console.error("Ajax call error when remove time sheet register.");
+    document.getElementById("alertTsheetDelRecord").style.display="block";
+    let messageArray = [];
+    for (const [index, key] of Object.entries(messages)) {
+        messageArray.push(window.i18nWlogTab[key]);
+    }
+    document.getElementById("alertTsheetDelRecord").innerHTML= messageArray.join('<br>');
```
Comment (adrianfish) [inline assignment/tool/src/webapp/js/studentViewSubmission.js:179]:
```suggestion
    document.getElementById("alertTsheetDelRecord").innerHTML
      = Object.entries(messages).reduce((acc, entry) => { acc.push(i18nWlogTab[entry[1]]); return acc; }, []).join("<br>");
```

**Review by adrianfish — DISMISSED** [review]
Some basic changes. You can ignore the reduce changes if you find them hard to read. I personally like Array.reduce as it helps make your variables immutable. More functional.

**Comment by feralvarez95** [comment]
@adrianfish It looks perfect for me too.
Thanks!

**Review by jesusmmp — APPROVED** [review]
LGTM!

**Review by juanjmerono — APPROVED** [review]

**Comment by feralvarez95** [comment]
I did a new commit correcting some Codacy issues that I introduced. All done @adrianfish!

**Pull request merged**